File: CC2 -  Finals (1Y 1S)/Factorial.java
/**
 * @author pao
 * BSCS 1B
 */
import java.util.Scanner;
public class Factorial {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String choice;
        int num;
        System.out.println("--------------------------");
        System.out.println("|  Factorial Calculator  |");
        System.out.println("--------------------------");
        do {
            long factorial = 1;
            System.out.print("Enter a positive integer: ");
            while (true) {
                if (sc.hasNextInt()) {
                    num = sc.nextInt();
                    sc.nextLine();
                    if (num >= 0) {
                        break;
                    } else {
                        System.out.print("Please enter a positive integer: ");
                    }
                } else {
                    System.out.println("Invalid input. Please enter a positive integer.");
                    sc.next();
                }
            }
            int loopCount = 1;
            while (loopCount <= num) {
                factorial *= loopCount;
                loopCount++;
            }
            System.out.println(num + "! = " + factorial);

            System.out.print("Do you want to calculate the factorial of another number? (Y/n): ");
            choice = sc.next();

        } while (!choice.equalsIgnoreCase("n"));
    }
}

File: Practice Codes/CC2 Java to Python/Finals/Looping/factorial.py
print(f"""|--------------------------------|
|      Factorial Calculator      |
|--------------------------------|""")
while True:
    number = int(input("Enter a number to calculate its factorial (-1 to exit): "))
    if number == -1:
        break
    elif number < 0:
        print("Factorial is not defined for negative numbers.")
    else:
        result = 1
        for i in range(1, number + 1):
            result *= i
        print(f"The factorial of {number} is: {result}")